classes/lib.php

```diff
@@ -59,4 +65,74 @@ public static function validate_ip_against_config() {
             send_unknown($msg);
         }
     }
+
+    /**
+     * Records that the cache was pinged. This is useful for cache debugging.
+     * @param int $previousincache
+     * @param int $previousindb
+     * @param int $new
+     * @param string $where cron or web
+     */
+    public static function record_cache_pinged(int $previousincache, int $previousindb, int $new, string $where) {
+        try {
+            $details = array_merge(
+                self::get_cache_log_details(),
+                [
+                    'previousvalueindb' => $previousindb,
+                    'previousvalueincache' => $previousincache,
+                    'newvalue' => $new,
+                    'where' => $where,
+                ]
+            );
+
+            $event = cache_ping::create([
+                'other' => $details,
+                'contextid' => context_system::instance()->id,
+            ]);
+            $event->trigger();
+        } catch (Throwable $e) {
+            debugging($e, DEBUG_DEVELOPER);
+        }
+    }
+
+    /**
+     * Records that the cache was checked. This is used for debugging cache mismatches
+     * @param int $valueincache
+     * @param int $valueindb
+     * @param string $where web or cron
+     */
+    public static function record_cache_checked(int $valueincache, int $valueindb, string $where) {
+        try {
+            $details = array_merge(
+                self::get_cache_log_details(),
+                [
+                    'valueindb' => $valueindb,
+                    'valueincache' => $valueincache,
+                    'where' => $where,
+                ]
+            );
+
+            $event = cache_check::create([
+                'other' => $details,
+                'contextid' => context_system::instance()->id,
+            ]);
+            $event->trigger();
+        } catch (Throwable $e) {
+            debugging($e, DEBUG_DEVELOPER);
+        }
+    }
+
+    /**
+     * Gets useful information about caches that can be logged when doing cache check ping
+     * @return array
+     */
+    private static function get_cache_log_details() {
+        global $CFG;
+
+        return [
+            'siteidentifier' => $CFG->siteidentifier ?? null,
+            'session_redis_prefix' => $CFG->session_redis_prefix ?? null,
```

Comment:
This is the session prefix not the cache prefix, its also tightly coupled to a muc store which doesn't feel right

Reply:
Have removed these after our discussion - not needed